main.go

```diff
@@ -105,14 +111,39 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
+	disableHTTP2 := func(c *tls.Config) {
+		if !enableHTTP2 {
+			c.NextProtos = []string{"http/1.1"}
+		}
+	}
+	if !enableHTTP2 {
+		setupLog.Info("Disabling HTTP2 support")
+	}
+
+	webhookServer := webhook.NewServer(webhook.Options{
+		Port:    9443,
+		TLSOpts: []func(*tls.Config){disableHTTP2},
+	})
+
+	metricsOptions := metricsserver.Options{
+		BindAddress: metricsAddr,
+		TLSOpts:     []func(*tls.Config){disableHTTP2},
+	}
```

Comment:
I find this a bit hard to read. I'd propose the following approach:

```suggestion
    disableHTTP2 := func(c *tls.Config) {
        c.NextProtos = []string{"http/1.1"} 
    }

    tlsOpts := []func(*tls.Config){}
    if !enableHTTP2 {
        tlsOpts = append(tlsOpts, disableHTTP2)
        setupLog.Info("Disabling HTTP2 support")
    }

    webhookServer := webhook.NewServer(webhook.Options{
		Port:    9443,
		TLSOpts: tlsOpts,
    })

	metricsOptions := metricsserver.Options{
		BindAddress: metricsAddr,
		TLSOpts:     tlsOpts,
	}
```

Otherwise, I'd say we can rename the `disableHTTP2` variable to something like `disableHTTP2IfRequested`. WDYT?

Reply:
Did some refactoring in 7f4c8a9c, take a look.

Reply:
Very nice!